webdriver/elements/active.py

```diff
@@ -0,0 +1,210 @@
+
+    assert_element_result(result)
+    assert get_id(session, result.body["value"]) == "document-body"
+    assert_dialog_handled(session, "accept #2")
+    assert read_global(session, "accept2"), True
+
+    create_dialog(session)("prompt", text="accept #3", result_var="accept3")
+
+    result = session.transport.send("GET",
+                                    "session/%s/element/active" % session.session_id)
+
+    assert_element_result(result)
+    assert get_id(session, result.body["value"]) == "document-body"
+    assert_dialog_handled(session, "accept #3")
+    assert read_global(session, "accept3") == ""
+
+# [...]
+# 2. Handle any user prompts and return its value if it is an error.
+# [...]
+# In order to handle any user prompts a remote end must take the following
+# steps:
+# 2. Run the substeps of the first matching user prompt handler:
+#
+#    [...]
+#    - missing value default state
+#    - not in the table of simple dialogs
+#      1. Dismiss the current user prompt.
+#      2. Return error with error code unexpected alert open.
+def test_handle_prompt_missing_value(session, create_dialog):
+    session.url = inline("<body id='document-body'><p>Hello, World!</p></body>")
+
+    create_dialog("alert", text="dismiss #1", result_var="dismiss1")
+
+    result = session.transport.send("GET",
+                                    "session/%s/element/active" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_dialog_handled(session, "dismiss #1")
+    assert session.execute_script("return accept1;") == None
+
+    create_dialog("confirm", text="dismiss #2", result_var="dismiss2")
+
+    result = session.transport.send("GET",
+                                    "session/%s/element/active" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_dialog_handled(session, "dismiss #2")
+    assert session.execute_script("return dismiss2;") == False
+
+    create_dialog("prompt", text="dismiss #3", result_var="dismiss3")
+
+    result = session.transport.send("GET",
+                                    "session/%s/element/active" % session.session_id)
+
+    assert_error(result, "unexpected alert open")
+    assert_dialog_handled(session, "dismiss #3")
+    assert session.execute_script("return dismiss3;") == None
+
+# > [...]
+# > 3. Let active element be the active element of the current browsing
+# >    context's document element.
+# > 4. Let active web element be the JSON Serialization of active element.
+# > 5. Return success with data active web element.
+def test_sucess_document(session):
+    session.url = inline("""
+        <body id="document-body">
+            <h1>Heading</h1>
+            <input id="the-input" />
+            <input id="interactable-input" />
+            <input id="non-interactable-input" style="opacity: 0;" />
+            <p>Another element</p>
+        </body>""")
```

Comment:
All the other items on the document are probably not relevant here, right? I get that you want to test that the first `<input>` is not selected by default but that feels like it’s testing HTML rather than WebDriver. HTML already defines that the default active element should be the document element, i.e. `<body>` in this case.

You probably want to test things like active element after interaction, frames &c. It’s of course fine to leave this until later.